src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,133 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Storage class to handle saving and loading tasks from a file
+ */
+public class Storage {
+    private File file;
+    private String filePath;
+
+    /**
+     * creates a Storage object
+     * @param filePath a String representing the directory of the file
+     */
+    Storage(String filePath) {
+        this.filePath = filePath;
+        file = new File(filePath);
+        if (!file.exists()) {
+            try {
+                file = new File("data");
+                file.mkdir();
+                file = new File("data/duke.txt");
+                file.createNewFile();
+            } catch (IOException e) {
+                System.out.println("Something went wrong: " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * saves tasks into a file
+     * @param filePath a String representing the directory of the file
+     * @param list list of tasks that will be written into the file
+     */
+    public void writeToFile(String filePath, TaskList list) {
+        try {
+            String text = "";
+            FileWriter fw = new FileWriter(filePath);
+            for (int i = 0; i < list.size(); i++) {
+                Task t = list.get(i);
+                text += textToAdd(t);
+            }
+            fw.write(text);
+            fw.close();
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    /**
+     * loads tasks from a file
+     * @param filePath a String representing the directory of the file
+     * @return return a TaskList object containing the list of tasks that was in the file
+     * @throws FileNotFoundException
+     */
+    public ArrayList<Task> readFile(String filePath) throws FileNotFoundException {
+        File f = new File(filePath);
+        Scanner s = new Scanner(f);
+        ArrayList<Task> list = new ArrayList<>();
+        while (s.hasNext()) {
+            String taskType = s.next();
+            switch(taskType) {
+            case "T":
+                String[] todoArr = s.nextLine().trim().split("\\|");
+                Task todo = new Todo(todoArr[2].trim());
+                if (Integer.parseInt(todoArr[1].trim()) == 1) {
+                    todo.markAsDone();
+                }
```

Comment:
There are many repeated instances of this. Perhaps, you could consider having an alternative constructor that already marks it as done?